toplevel/coqloop.ml

```diff
@@ -469,6 +469,16 @@ let rec vernac_loop ~state =
   (* Exception printing should be done by the feedback listener,
      however this is not yet ready so we rely on the exception for
      now. *)
+  | Sys_blocked_io ->
+    (* the parser doesn't like nonblocking mode, cf #10918 *)
+    let msg =
+      Pp.(str "Uncaught exception Sys_blocked_io." ++ spc() ++
+          str "If stdin is in non-blocking mode please reset it to blocking mode,"
+          ++ spc() ++ str "otherwise please report at "
+          ++ str Coq_config.wwwbugtracker ++ str".")
+    in
+    TopErr.print_error_for_buffer Feedback.Error msg top_buffer;
+    exit 1
```

Comment:
Should we instead register this exception printer in the regular way ?

Reply:
I don't think this message makes sense elsewhere.

Reply:
We can register it this module then?

Reply:
I don't think having a special case is worth it, at some point errors should be sealed so the less special cases we have the better, specially when this error is so esoteric.

Reply:
We need a special case to exit, otherwise it will go back in the loop.

Reply:
The message you print makes me think this should be an anomaly?

Reply:
But indeed we have no way yet to register those.

Reply:
It's only an anomaly if it was triggered in some way other than stdin being nonblocking. I don't know if that's possible.

Reply:
No idea; looking at the code it can use some refactoring; I'll push another PR.

Reply:
May I suggest the following message:
```
Coqtop cannot work properly since the standard input is in non-blocking mode.
Use <command> to clear the non-blocking flag.
```

Reply:
No bugreport, no uncaught exn...

Reply:
@SkySkimmer are you planning to implement this suggestion? Indeed the proposed message looks better to me.

Reply:
What's `<command>`?

Reply:
https://unix.stackexchange.com/a/558108/45323 suggests
```bash
perl -MFcntl -e 'fcntl STDIN, F_SETFL, fcntl(STDIN, F_GETFL, 0) & ~O_NONBLOCK'
```

Reply:
Maybe there's an equivalent we can give with `ocamlrun` or something instead?

Reply:
https://unix.stackexchange.com/a/450066/45323 suggests that running `bash` and then executing `ctrl+D` inside the inner bash is enough

Reply:
https://stackoverflow.com/a/48011063/377022 suggests
```bash
python3 -c $'import os\nos.set_blocking(0, True)'
```

Reply:
`stty sane` (to be tested)